bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java

```diff
@@ -121,19 +120,20 @@ public static byte getResponseHeader(byte[] response) {
     }
 
     public byte[] getRequest() {
-        byte[] value = ArrayUtils.EMPTY_BYTE_ARRAY;
-        value = ArrayUtils.add(value, HEADER_PREFIX);
-        value = ArrayUtils.add(value, header);
-        value = ArrayUtils.add(value, (byte) data.length);
-        value = ArrayUtils.addAll(value, data);
-        value = ArrayUtils.add(value, createChecksum(value));
-        return ArrayUtils.addAll(REQUEST_PREFIX, value);
+        byte[] value = new byte[4 + data.length + REQUEST_PREFIX.length];
+        System.arraycopy(REQUEST_PREFIX, 0, value, 0, REQUEST_PREFIX.length);
+        value[REQUEST_PREFIX.length] = HEADER_PREFIX;
+        value[REQUEST_PREFIX.length + 1] = header;
+        value[REQUEST_PREFIX.length + 2] = (byte) data.length;
+        System.arraycopy(data, 0, value, REQUEST_PREFIX.length + 3, data.length);
+        value[value.length-1] = createChecksum(value, REQUEST_PREFIX.length, REQUEST_PREFIX.length + 2);
+        return value;
     }
 
-    protected byte createChecksum(byte[] data) {
+    protected byte createChecksum(byte[] data, int startIndex, int maxIndex) {
```

Comment:
`maxIndes` is probably an inappropriate name as in fact it is the first index that will not be considered.
At least it should be commented.

Reply:
i added javadoc, changed from maxindex to length. should be more clear now. thanks!